proposals/exp-show-target.md

```diff
@@ -0,0 +1,214 @@
+```
+
+This table view is from a simplified example and is still missing several
+columns, so this view can become cluttered quickly. Eliminating irrelevant
+information is a priority.
+
+There are several ways in which the included parameters might be irrelevant:
+* Parameters files may include values that aren't tracked by DVC at all (for
+  example, debug level).
+* The parameters may be used in a different pipeline branch (for example, the
+  `topic_modeling` stage is in a separate branch of the pipeline from the
+  `evaluate` stage, which produces the relevant experiment metrics).
+* The parameters may be used in a downstream pipeline branch (for example, the
+  post-processing `combine` stage parameters aren't relevant to experiment
+  evaluation).
+
+```console
+                +-------------------+
+                | data/data.xml.dvc |
+                +-------------------+
+                          *
+                          *
+                          *
+                     +---------+
+                     | prepare |
+                     +---------+
+                          *
+                          *
+                          *
+                    +-----------+
+                    | featurize |
+                  **+-----------+**
+              ****       *         ****
+          ****           *             ***
+        **              *                 ***
++-------+               *                    **
+| train |             **                      *
++-------+            *                        *
+         **        **                         *
+           **    **                           *
+             *  *                             *
+        +----------+                  +----------------+
+        | evaluate |                  | topic_modeling |
+        +----------+                  +----------------+
+                   ***             ****
+                      *         ***
+                       **     **
+                     +---------+
+                     | combine |
+                     +---------+
+```
+
+There are no apparent reasons that users want to see untracked parameters in
+`dvc exp show`. Unlike `dvc params diff` and `dvc metrics diff`, `dvc exp show`
+is not applicable to arbitrary files. Likewise, there are no apparent reasons
+that users want to see parameters or metrics from irrelevant stages of their
+pipelines. Users could keep untracked parameters in other files if they don't
+want them shown in `dvc exp show`, and they could exclude parameters or metrics
+from irrelevant stages, but DVC already has enough information to understand
+that these should not be shown.
+
+# Detailed design
+
+`dvc exp show` could take stages as (positional?) arguments, similar to `dvc exp
+run`. The help message could be similar:
+
+```console
+positional arguments:
+stages               Stages for which to print experiment info. 'dvc.yaml'
+by default.
```

Comment:
I'd go for targets for consistency with existing commands (e.g. [`repro`](https://dvc.org/doc/command-reference/repro)), and since dvc.yaml files can be targets too.

```suggestion
usage: dvc experiments show [-h] [-q | -v] [-a] [-T] [-A] [-n <num>]
                            ...
                            [targets [<target> ...]]

targets              Stages to include. 'dvc.yaml' by default.
```

Reply:
What about `dvc params/metrics diff --targets`, which use `targets` to mean filenames and not stages?

Reply:
I'm thinking of existing commands that target stages (again like `repro` which predate params and metrics BTW). But yeah `--targets` for some reason was designed to specifically accept file names and not param/metric stages, maybe something else to reconsider but seems unrelated here 🙂